central/networkgraph/flow/datastore/internal/store/postgres/flow_impl_benchmark_test.go

```diff
@@ -0,0 +1,183 @@
+	b.Run("benchmark upsert 100 flow batch", benchmarkUpsertFlows(flowStore, 100))
+	b.Run("benchmark upsert 1000 flow batch", benchmarkUpsertFlows(flowStore, 1000))
+}
+
+func benchmarkUpsertFlows(flowStore store.FlowStore, numFlows uint32) func(*testing.B) {
+	return func(b *testing.B) {
+		flows := make([]*storage.NetworkFlow, 0, numFlows)
+		for i := uint32(0); i < numFlows; i++ {
+			id, err := testutils.ExtIdFromIPv4(fixtureconsts.Cluster1, i)
+			require.NoError(b, err)
+			flows = append(flows, testutils.ExtFlow(fixtureconsts.Deployment1, id.String()))
+		}
+
+		b.ResetTimer()
+
+		for i := 0; i < b.N; i++ {
+			err := flowStore.UpsertFlows(context.Background(), flows, timestamp.Now()-1000000)
+			require.NoError(b, err)
+
+			exclude(b, func() {
+				err = flowStore.RemoveFlowsForDeployment(context.Background(), fixtureconsts.Deployment1)
+				require.NoError(b, err)
+			})
+		}
+	}
+}
+
+func BenchmarkGetExternalFlows(b *testing.B) {
+	psql := pgtest.ForT(b)
+
+	clusterStore := NewClusterStore(psql)
+	flowStore, err := clusterStore.CreateFlowStore(context.Background(), fixtureconsts.Cluster1)
+	require.NoError(b, err)
+
+	setupExternalIngressFlows(b, flowStore, fixtureconsts.Deployment1, 1000)
+	setupExternalIngressFlows(b, flowStore, fixtureconsts.Deployment2, 10000)
+
+	setupExternalIngressFlows(b, flowStore, fixtureconsts.Deployment3, 1000)
+	setupExternalEgressFlows(b, flowStore, fixtureconsts.Deployment3, 1000)
+
+	b.Run("deployment with 1000 external flows", benchmarkGetExternalFlows(flowStore, fixtureconsts.Deployment1))
+	b.Run("deployment with 10000 external flows", benchmarkGetExternalFlows(flowStore, fixtureconsts.Deployment2))
+	b.Run("deployment with 1000 ingress and 1000 egress flows", benchmarkGetExternalFlows(flowStore, fixtureconsts.Deployment3))
+}
+
+func BenchmarkGetFlowsForDeployment(b *testing.B) {
+	psql := pgtest.ForT(b)
+
+	clusterStore := NewClusterStore(psql)
+	flowStore, err := clusterStore.CreateFlowStore(context.Background(), fixtureconsts.Cluster1)
+	require.NoError(b, err)
+
+	// 1000 flows 1 -> 2
+	setupDeploymentFlows(b, flowStore, fixtureconsts.Deployment1, fixtureconsts.Deployment2, 1000)
+	// 1000 flows 2 -> 4 (2 == 2000 flows total)
+	setupDeploymentFlows(b, flowStore, fixtureconsts.Deployment2, fixtureconsts.Deployment4, 1000)
+	// 10000 flows 3 -> 4 (4 == 11000 flows total)
+	setupDeploymentFlows(b, flowStore, fixtureconsts.Deployment3, fixtureconsts.Deployment4, 10000)
+
+	b.Run("deployment with 1000 flows", benchmarkGetFlowsForDeployment(flowStore, fixtureconsts.Deployment1))
+	b.Run("deployment with 10000 flows", benchmarkGetFlowsForDeployment(flowStore, fixtureconsts.Deployment3))
+	b.Run("deployment with 1000 ingress and 1000 egress flows", benchmarkGetFlowsForDeployment(flowStore, fixtureconsts.Deployment2))
+}
+
+func setupDeploymentFlows(b *testing.B, flowStore store.FlowStore, fromId string, toId string, numFlows int) {
+	flows := make([]*storage.NetworkFlow, 0, numFlows)
```

Comment:
[Nit] I prefer declaring the size of the array and assigning to the elements, rather than declaring the capacity and using append.

Reply:
I've left these as-is since it's a stylistic change and not a functional one